pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -31,15 +31,21 @@ import (
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/clock"
 	"k8s.io/utils/pointer"
 )
 
-// Now determines the current metav1.Time.
-var Now = metav1.Now
+// Clock defines the clock for the helper functions
+var Clock clock.Clock = clock.RealClock{}
```

Comment:
Why do we still need this global clock? Wouldn't it be cleaner to always pass a clock to the functions instead of using a global var?

Reply:
There are some controllers left (seed care, shoot care, legacy controller of gardenlet) which do not have a clock yet. For those I kept the global clock.

Reply:
OK, can you add a doc string that this should not be used for new code and is deprecated and will be removed soon?

Reply:
Or do you want to replace the remaining usages right away now? How much effort it is? If it's too much let's keep it for now (we have to touch this code anyways when we do the refactoring to controller-runtime controllers soon).

Reply:
I added the deprecation messages to the old functions now.
They are used quite a few times. When you follow the tree, the usages end up in gardenlet legacy controllers and resource manager. Both do not have a `clock.Clock`.
Imho we should replace the usages when we refactor those controllers rather then touch them twice.